cmd/kubectl-gadget/version.go

```diff
@@ -36,6 +40,46 @@ var versionCmd = &cobra.Command{
 	Use:   "version",
 	Short: "Show version",
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println(version)
+		fmt.Println("Client version:", version)
+
+		client, err := k8sutil.NewClientsetFromConfigFlags(utils.KubernetesConfigFlags)
+		if err != nil {
+			fmt.Println("Error while creating a new clientset:", err)
+			return
+		}
+
+		opts := metav1.ListOptions{LabelSelector: "k8s-app=gadget"}
+		pods, err := client.CoreV1().Pods("gadget").List(context.TODO(), opts)
+		if err != nil {
+			fmt.Println("Error while getting the gadget pods:", err)
+			return
+		}
+
+		serverVersions := make(map[string]struct{})
+		for _, pod := range pods.Items {
+			image := pod.Spec.Containers[0].Image
+
+			// Get the image tag
+			parts := strings.Split(image, ":")
+			if len(parts) != 2 {
+				continue
+			}
```

Comment:
This approach fails to extract the image tag in my development environment:

```bash
$ ./kubectl-gadget version
Client version: v0.8.0-55-g4289e7627b24-dirty
Server version: not installed
```

It is because I am using a local registry at `192.168.1.105:5000`, so there are two `:` 😅 :

```bash
$ k get pods -l k8s-app=gadget -n gadget -o jsonpath={.items[0].spec.containers[0].image}
192.168.1.105:5000/gadget:version
```

I think you can verify that `parts > 0`, and simply consider that `parts[len(parts)-1]` is the image tag. What do you think?

Reply:
I think that parts[len(parts)-1] would be always the tag. (I am no Docker expert :D )

So yes, i would agree